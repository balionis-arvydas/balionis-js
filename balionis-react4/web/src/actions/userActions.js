import axios from '../services/axiosService';

const rest = axios.create();

export const readUser = async (userId) => {
    const request = rest.get('/v1/user/' + userId);

    const message = await request
        .then(res => JSON.stringify(res.data))
        .catch(error => "{ \"reason\": \"" + JSON.stringify(error.message) + "\"}");

    return message;
};
