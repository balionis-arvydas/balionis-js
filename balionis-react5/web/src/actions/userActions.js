import axios from '../services/axiosService';
import { getSocket } from '../services/socketService';
import logger from '../services/logger';

const rest = axios.create();
const log = logger.create('userActions');

export const readUserWithRest = async (userId) => {
    const url = '/v1/user/' + userId;
    log.debug('readUser:', 'url=' + url);

    const request = rest.get(url);

    const message = await request
        .then(res => {
            const data = JSON.stringify(res.data);
            log.debug('readUser:', 'data=' + data);
            return data;
        }).catch(error => {
            log.warn('readUser:', 'error=' + error);
            return "{ \"reason\": \"" + JSON.stringify(error.message) + "\"}";
        });

    return message;
};

export const readUserWithSocket = async (userId) => {
    log.debug("readUser:", `userId=${userId}`);

    const promise = new Promise(resolve => {
        const socket = getSocket();
        socket.emit("getUser", { userId });

        const handler = async (message) => {
            socket.off("getUser", handler);
            resolve(message);
        };
        socket.on("getUser", handler);
    });

    return promise;
};

export const readUser = async (userId, useSocket) => {
    const fn = useSocket ? readUserWithSocket : readUserWithRest;
    return fn(userId);
};
