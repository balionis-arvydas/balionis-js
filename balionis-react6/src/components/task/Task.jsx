import React from "react";
import { connect } from "react-redux";

import { mapStateToProps, mapDispatchToProps } from "./TaskProperties";

import "./Task.less";

class Task extends React.Component {

    constructor (props) {
        super(props);
    }

    handleClick (event) {
        event.preventDefault();
        this.props.removeTask(this.props.task.id);
    }

    render () {
        return (
            <li className="taskWrapper" onClick={(event) => this.handleClick(event)}>
                <span>{this.props.task.content}</span>
            </li>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Task);
