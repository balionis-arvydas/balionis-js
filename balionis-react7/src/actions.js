import c from "./constants";
import helpers from "./helpers";

const open = async (testURL, options = {}) => {
    const page = await browser.newPage();
    const url = testURL || helpers.getGlobalHomeURL();
    await page.goto(url, options);
    return page;
};

const login = async (testURL, waitForRedirect) => {
    jest.setTimeout(c.LOGIN_TIMEOUT);
    const page = await open(testURL, { timeout: c.LOGIN_TIMEOUT });

    console.info("login: url=" + page.url());

    if (waitForRedirect) {
        await page.waitForNavigation();
    }

    await page.waitForSelector(c.LOGIN_SELECTOR, { timeout: c.LOGIN_TIMEOUT });

    return page;
};

export default {
    open,
    login,
}
