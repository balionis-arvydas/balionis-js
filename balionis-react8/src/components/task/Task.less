// @import "colors.less";
.taskWrapper {
    border: 1px solid #00ff00;
}
