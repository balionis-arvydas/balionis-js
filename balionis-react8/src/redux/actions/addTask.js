import { ADD_TASK } from "./actionTypes";

let nextTaskId = 0;

export const addTask = (content, taskId) => ({
    type: ADD_TASK,
    payload: {
        id: (taskId ? taskId : ++nextTaskId),
        content,
    }
});
