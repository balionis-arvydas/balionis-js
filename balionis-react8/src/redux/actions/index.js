export * from './addTask';
export * from './removeTask';
