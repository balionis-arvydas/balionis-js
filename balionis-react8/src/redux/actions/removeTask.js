import { REMOVE_TASK } from "./actionTypes";

export const removeTask = id => ({
    type: REMOVE_TASK,
    payload: {
        id
    }
});
