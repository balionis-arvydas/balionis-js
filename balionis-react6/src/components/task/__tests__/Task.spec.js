import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore } from '../../../redux/__tests__/mockStore';

import Task from '../Task';
import { removeTask } from "../../../redux/actions";

describe('TaskComponent', () => {
    const task = {
        id: 123,
        content: "ABC",
    };

    const props = {
        task,
    };

    let store;

    beforeAll(() => {
        jest.resetAllMocks();
    });

    beforeEach(() => {
        store = mockStore({
            taskList: { tasks: [] }
        });
        store.dispatch = jest.fn();
    });

    test("snapshot", async () => {
        const component = renderer.create(
            <Provider store={store}>
                <Task task={task}/>,
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test("should render content", async () => {
        const component = mount(
            <Provider store={store}>
                <Task {...props}/>
            </Provider>
        );
        expect(component.find('span').text()).toEqual('ABC');
    });

    test("should remove task", async () => {
        const component = mount(
            <Provider store={store}>
                <Task {...props}/>
            </Provider>
        );

        component.find('li').simulate('click');

        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            removeTask(123)
        );
    });
});
