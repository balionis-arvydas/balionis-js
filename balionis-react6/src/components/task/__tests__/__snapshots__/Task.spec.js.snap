// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskComponent snapshot 1`] = `
Array [
  <li
    className="taskWrapper"
    onClick={[Function]}
  >
    <span>
      ABC
    </span>
  </li>,
  ",",
]
`;
